### PR TITLE
Removed config keys for <table>Plugin Now use the standard dot notation.

### PR DESCRIPTION
**Pull request by mtancoigne**
As I can't make the tests yet, let's Travis do this...

### CODE REVIEW

**Comment by mtancoigne** [comment]
Great... I'll look at this tomorrow...

**Comment by dereuromark** [comment]
Wow, doesn't that look way cleaner and easier?

File: src/Auth/TinyAuthorize.php [hunk 1]
```diff
@@ -362,24 +355,12 @@ protected function _getUserRoles($user) {
...
+		// Multi-role case : load the pivot table
 		$pivotTableName = $this->_config['pivotTable'];
-		$usersTableName = $this->_config['usersTable'];
 		if (!$pivotTableName) {
-			$tables = [
-				$usersTableName,
-				$rolesTableName
-			];
-			asort($tables);
-			$pivotTableName = implode('', $tables);
```
Comment (dereuromark) [inline src/Auth/TinyAuthorize.php]: Did you remove the autodetect here? In case no config is provided? Should we?

**Comment by mtancoigne** [comment]
I'copy/paste the discussion with @bravo-kernel wich bring me here:

```irc
[22:17:00 CET] <bravo_kernel> we keep the "old" rolesTable setting
[22:17:13 CET] <elcms> yep
[22:17:27 CET] <bravo_kernel> make it support either Roles or Plugin.Roles inside the code
[22:17:47 CET] <elcms> so we split it in our code
[22:17:50 CET] <bravo_kernel> same for the other two "old" variables
[22:18:05 CET] <bravo_kernel> exactly, whatever is needed there
[22:18:31 CET] <bravo_kernel> the plugin syntax is 100% trustworthy so if detected we try to load a plugin class
[22:18:34 CET] <elcms> yeah, that sound a great way to keep it right with cake
[22:18:44 CET] <bravo_kernel> I think it is
[22:18:55 CET] <bravo_kernel> just for fun
[22:19:04 CET] <bravo_kernel> I learned Cake3 updating this plugin from cake2
[22:19:08 CET] <elcms> now there's something : if the pivot table is in anther plugin ?
[22:19:16 CET] <bravo_kernel> the hardest part was grasping the plugin notation :P
[22:19:29 CET] <bravo_kernel> ok
[22:19:37 CET] <elcms> bah, the user should fill the right thing
[22:19:40 CET] <bravo_kernel> hmmm
[22:19:48 CET] <bravo_kernel> probably 
[22:19:50 CET] <elcms> and we should stop to construct it by ourselve
[22:19:55 CET] <bravo_kernel> and maybe we should not overengineer
[22:19:57 CET] <elcms> *ourselves
[22:20:18 CET] <elcms> I agree
[22:20:34 CET] <bravo_kernel> I would say.... detect plugin notation: try loading class from plugin
[22:20:43 CET] <bravo_kernel> no plugin: load plain
[22:20:55 CET] <bravo_kernel> seems like that could cater all possibilities or not?
[22:21:11 CET] <elcms> explode the chain, if more than 2 items in the array, we have a plugin, that's ok
[22:21:23 CET] <bravo_kernel> haha
[22:21:45 CET] <bravo_kernel> I can't tell for sure
[22:21:58 CET] <bravo_kernel> but you will figure it out, maybe check the core somewhere to see how they do it
[22:22:09 CET] <bravo_kernel> I'm only a hobbyist so far from a pro
[22:22:14 CET] <elcms> Plugin.Model is basically two strings with a dot
[22:22:24 CET] <bravo_kernel> totally true :)
[22:22:41 CET] <bravo_kernel> I don't think you really can go wrong there
[22:22:48 CET] <elcms> same here, but i'm a nearby full free time hobbyist
[22:22:53 CET] <bravo_kernel> and Mark will jump in if needed but this feels right
[22:23:01 CET] <bravo_kernel> haha, not unplanned I hope
[22:23:12 CET] <elcms> ok, let sum this up
[22:23:40 CET] <elcms> we keep usersTable, rolesTable, pivotTable
[22:23:51 CET] <bravo_kernel> check
[22:24:17 CET] <elcms> if pivotTable is empty and we are in multi-group, throw an error for misconfiguration
[22:24:55 CET] <elcms> else, do as natural with the given config : test the strings and load the models
[22:25:18 CET] <bravo_kernel> sounds good to me
[22:25:24 CET] <bravo_kernel> i like exceptions for misconfigurations
[22:25:40 CET] <elcms> me too
[22:25:44 CET] <bravo_kernel> cool :)
```

TL;DR; we get rid of autoconfig and throw an exception if the pivot table is missing, so we don't even need to parse the table name for a plugin in it.

File: src/Auth/TinyAuthorize.php (hunk 1 above)
Comment (bravo-kernel) [inline src/Auth/TinyAuthorize.php]: I personally prefer the exception and less magic,

**Comment by bravo-kernel** [comment]
@mtancoigne could you please try to prevent pasting personal info into public resource?

Final question about this PR; does this work for namespaced plugins as well? E.g. MyCorp\Plugin.TableClass?

File: src/Auth/TinyAuthorize.php [hunk 2]
```diff
@@ -362,24 +355,12 @@ protected function _getUserRoles($user) {
...
+			throw new Exception('Missing TinyAuthorize pivotTable. Check your configuration');
```
Comment (bravo-kernel) [inline src/Auth/TinyAuthorize.php]: This message make me think the table does not exist (in the database). Maybe change it to something like below, would make it consistent with the other feedback as well.

> Invalid TinyAuthorize MultiRole Setup (no pivot table found in Configure)

Comment (mtancoigne) [inline src/Auth/TinyAuthorize.php]: Sorry for the paste...

Comment (dereuromark) [inline src/Auth/TinyAuthorize.php]: I agree. But since we have that logic here why do we not use conventions to find it? The error will then come from the orm and will say exaxtly what it said so far.

Comment (bravo-kernel) [inline src/Auth/TinyAuthorize.php]: @dereuromark I guess it's a preference thing. I always appreciate being informed to the root cause concrete as possible and this would point me straight to Tiny's config but I can live with the ORM exception as well. Your plugin... your call :relaxed:

Comment (dereuromark) [inline src/Auth/TinyAuthorize.php]: Since its cached performance is not an issue :)
I like convention driven. But maybe later
